src/main/java/LinkedList.java

```diff
@@ -10,47 +12,41 @@ public LinkedList() {
 	/* Constructor that creates list of size with nodes that represent their index */
 	public LinkedList(int size) {
 		for(int i = 0; i < size; i++) {
-			add(i);
+			add(new ListNode(i));
 		}
 	}
 	
-	public void add(int number) {
-		addNode(new ListNode(number));
-	}
-	
-	private void addNode(ListNode node) {
+	public void add(ListNode node) {
 		if(head == null && tail == null) { //Empty LinkedList case
 			head = node;
-			tail = head;
+			tail = node;
 			return;
 		}
 		tail.next = node;
 		tail = tail.next;
 	}
 	
-	public int pop() {
-		return popNode().data;		
-	}
-	
-	private ListNode popNode() {
+	public ListNode pop() {
```

Comment:
Minor point - for lists, you typically have add/remove/first instead of push/pop/peek